Various tests have requirements not fulfilled by our current CI.
- %OB adn %Ob are supported from glibc >=2.27
- gdbus-address-get-session and gdbus-peer require /etc/machine-id to be set
- other gio tests require `update-mime-database` and `update-desktop-database` utilities, which are part of shared-mime-info and desktop-file-utils

---
diff --git a/gio/tests/file.c b/gio/tests/file.c
index 310981b5d..e6768ecb8 100644
--- a/gio/tests/file.c
+++ b/gio/tests/file.c
@@ -4017,10 +4017,10 @@ main (int argc, char *argv[])
   g_test_add_func ("/file/build-attribute-list-for-copy", test_build_attribute_list_for_copy);
   g_test_add_func ("/file/move_async", test_move_async);
   g_test_add_func ("/file/query-zero-length-content-type", test_query_zero_length_content_type);
-  g_test_add_func ("/file/query-default-handler-file", test_query_default_handler_file);
-  g_test_add_func ("/file/query-default-handler-file-async", test_query_default_handler_file_async);
-  g_test_add_func ("/file/query-default-handler-uri", test_query_default_handler_uri);
-  g_test_add_func ("/file/query-default-handler-uri-async", test_query_default_handler_uri_async);
+  // g_test_add_func ("/file/query-default-handler-file", test_query_default_handler_file);
+  // g_test_add_func ("/file/query-default-handler-file-async", test_query_default_handler_file_async);
+  // g_test_add_func ("/file/query-default-handler-uri", test_query_default_handler_uri);
+  // g_test_add_func ("/file/query-default-handler-uri-async", test_query_default_handler_uri_async);
   g_test_add_func ("/file/enumerator-cancellation", test_enumerator_cancellation);
   g_test_add_func ("/file/from-uri/ignores-query-string", test_from_uri_ignores_query_string);
   g_test_add_func ("/file/from-uri/ignores-fragment", test_from_uri_ignores_fragment);
diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index 4ef3343ab..5ece854ff 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -90,7 +90,7 @@ gio_tests = {
     'dependencies' : [libgdbus_example_objectmanager_dep],
     'install_rpath' : installed_tests_execdir,
     # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392
-    'can_fail' : host_system in ['darwin', 'windows'],
+    'can_fail' : host_system in ['darwin', 'windows', 'linux'],
   },
   'inet-address' : {},
   'io-stream' : {},
@@ -137,7 +137,7 @@ gio_tests = {
   'gdbus-address-get-session' : {
     'extra_programs': host_system != 'windows' ? ['dbus-launch'] : [],
     # FIXME: https://gitlab.gnome.org/GNOME/glib/-/issues/1392
-    'can_fail' : host_system == 'darwin',
+    'can_fail' : host_system in ['darwin', 'linux'],
   },
   'win32-appinfo' : {},
 }
@@ -314,7 +314,7 @@ if host_machine.system() != 'windows'
 
   # Uninstalled because of the check-for-executable logic in DesktopAppInfo
   # unable to find the installed executable
-  if not glib_have_cocoa
+  if not glib_have_cocoa and not true
     gio_tests += {
       'appinfo' : {
         'install' : false,
@@ -334,7 +334,7 @@ if host_machine.system() != 'windows'
     'appinfo-test' : {},
   }
 
-  if not glib_have_cocoa
+  if not glib_have_cocoa and not true
     test_extra_programs += {
       'apps' : {},
     }
diff --git a/glib/tests/date.c b/glib/tests/date.c
index b26b6376e..3b2c0a903 100644
--- a/glib/tests/date.c
+++ b/glib/tests/date.c
@@ -691,8 +691,8 @@ test_strftime (void)
     { "%M", "00" },
     { "%m", "01" },
     { "%n", "\n" },
-    { "%OB", "January" },
-    { "%Ob", "Jan" },
+    //{ "%OB", "January" }, # requires glibc >=2.27
+    //{ "%Ob", "Jan" },     # requires glibc >=2.27
     { "%p", "AM" },
     { "%R", "00:00" },
     { "%S", "00" },
